conda/models/dist.py

```diff
@@ -36,41 +41,61 @@ def __call__(cls, *args, **kwargs):
                 return Dist.from_string(value.repodata_record.fn,
                                         channel_override=value.channel.canonical_name)
             elif isinstance(value, Channel):
-                return Dist.from_url(value.url())
+                raise NotImplementedError()
             else:
                 return Dist.from_string(value)
         else:
             return super(DistType, cls).__call__(*args, **kwargs)
 
+    def __new__(cls, name, bases, dct):
+         dct['__slots__'] = ('channel', 'name', 'version', 'build_string', 'build_number', 'with_features_depends')
+         return type.__new__(cls, name, bases, dct)
+
 
 @with_metaclass(DistType)
-class Dist(Entity):
-
-    channel = StringField(required=False, nullable=True, immutable=True)
-
-    dist_name = StringField(immutable=True)
-    name = StringField(immutable=True)
-    version = StringField(immutable=True)
-    build_string = StringField(immutable=True)
-    build_number = IntegerField(immutable=True)
-
-    with_features_depends = StringField(required=False, nullable=True, immutable=True)
-    base_url = StringField(required=False, nullable=True, immutable=True)
-    platform = StringField(required=False, nullable=True, immutable=True)
-
-    def __init__(self, channel, dist_name=None, name=None, version=None, build_string=None,
-                 build_number=None, with_features_depends=None, base_url=None, platform=None):
-        # if name is None:
-        #     import pdb; pdb.set_trace()
-        super(Dist, self).__init__(channel=channel,
-                                   dist_name=dist_name,
-                                   name=name,
-                                   version=version,
-                                   build_string=build_string,
-                                   build_number=build_number,
-                                   with_features_depends=with_features_depends,
-                                   base_url=base_url,
-                                   platform=platform)
+class Dist(object):
+
+    def __init__(self, channel, name=None, version=None, build_string=None, build_number=None,
```

Comment:
This is going in the _opposite_ direction from the way we should be going. This is _increasing_ the dependence on the Dist object to determine name/version/build. We should be moving _away_ from that. Dist should _not_ have `name`, `version`, and `build_string` in it at all, and we _certainly_ should not be constructing the filename from it. The filename is what it is, and it not ought be required to follow this convention.